frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/breakdown-card/utils.tsx

```diff
@@ -0,0 +1,104 @@
+import * as _ from 'lodash';
+import { Humanize } from '@console/internal/components/utils';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+import { Colors } from './bar-colors';
+
+const getTotal = (stats: StackDataPoint[]) => {
+  return stats.reduce((total, dataPoint) => total + dataPoint.y, 0);
+};
+
+const addOthers = (stats: StackDataPoint[], totalUsed: string, humanize: Humanize) => {
+  const top5Total = getTotal(stats);
+  const others = Number(totalUsed) - top5Total;
+  const othersData = {
+    x: '',
+    y: others,
+    name: 'Others',
+    color: Colors.OTHER,
+    label: humanize(others).string,
+  };
+  return othersData;
+};
+
+export const addAvailable = (
+  stats: StackDataPoint[],
+  total: string,
+  used: string,
+  totalUsed: string,
+  humanize: Humanize,
+) => {
+  const availableInBytes = Number(total) - Number(used);
+  let othersData = {};
+  if (stats.length === 5) {
+    othersData = addOthers(stats, totalUsed, humanize);
+  }
+  const availableData = {
+    x: '',
+    y: availableInBytes,
+    label: humanize(availableInBytes).string,
+  };
+  return _.isEmpty(othersData) ? [...stats, availableData] : [...stats, othersData, availableData];
+};
+
+export const getBarRadius = (index: number, length: number) => {
+  let barRadius = {};
+  if (index === 0) {
+    barRadius = { bottom: 3 };
+  }
+  if (index === length - 1) {
+    barRadius = { top: 3 };
+  }
+  return barRadius;
+};
+
+export const isAvailableBar = (index: number, length: number) => {
+  let barColor = {};
+  if (index === length - 1) {
+    barColor = { fill: Colors.AVAILABLE };
+  }
+  return barColor;
+};
+
+export const getStackChartStats: GetStackStats = (response, metric, humanize) => {
+  const results = _.get(response, 'data.result', []);
+  return results.map((r) => {
+    const y = parseFloat(_.get(r, 'value[1]'));
+    const name = _.get(r, ['metric', metric], '');
+    const capacity = humanize(y).string;
+    return {
+      // INFO: x value needs to be same for single bar stack chart
+      x: '',
+      y,
+      name: _.truncate(name, { length: 12 }),
+      link: name,
+      color: Colors.LINK,
+      label: capacity,
+    };
+  });
+};
+
+type GetStackStats = (
+  response: PrometheusResponse,
+  metric: string,
+  humanize: Humanize,
+) => StackDataPoint[];
+
+export type StackDataPoint = {
+  x: string;
+  y: number;
+  name: string;
+  label: string;
+  link: string;
+  color: string;
+};
+
+export const getCapacityValue = (cephUsed: any, cephTotal: any, humanize: Humanize) => {
```

Comment:
`cephUsed` and `cephTotal` should have better types than `any`.
missing return type

Reply:
ack